pkg/config/logs/log.go

```diff
@@ -259,6 +259,35 @@ func (s *logWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+const tlsHandshakeErrorPrefix = "http: TLS handshake error"
+
+// tlsHandshakeErrorWriter writes TLS handshake errors to log with
+// debug level, to avoid flooding of tls handshake errors.
+type tlsHandshakeErrorWriter struct {
```

Comment:
I think the name of this struct can be a bit confusing. Would it make sense to rename it ? Maybe the name could be a bit more generic.